tests/test_anno.py

```diff
@@ -0,0 +1,72 @@
+from .mockers import (GitRepositorySandbox, assert_command, launch_command)
+
+
+class TestAnno:
+
+    def setup_method(self) -> None:
+
+        self.repo_sandbox = GitRepositorySandbox()
+
+        (
+            self.repo_sandbox
+            # Create the remote and sandbox repos, chdir into sandbox repo
+            .new_repo(self.repo_sandbox.remote_path, "--bare")
+            .new_repo(self.repo_sandbox.local_path)
+            .execute(f"git remote add origin {self.repo_sandbox.remote_path}")
+            .execute('git config user.email "tester@test.com"')
+            .execute('git config user.name "Tester Test"')
+        )
+
+    def test_anno(self) -> None:
+        """
+        Verify behaviour of a 'git machete anno' command.
+        """
+
+        (
+            self.repo_sandbox.new_branch("master")
+                .commit("master commit.")
+                .new_branch("develop")
+                .commit("develop commit.")
+                .new_branch("feature")
+                .commit("feature commit.")
+                .check_out("develop")
+                .commit("New commit on develop")
+        )
+        launch_command('discover', '-y')
+
+        # Test `git machete anno` without providing the branch name
+        launch_command('anno', 'Custom annotation for `develop` branch')
+        assert_command(
+            ["status"],
+            """
+            master (untracked)
+
+            develop *  Custom annotation for `develop` branch (untracked)
+            |
+            x-feature (untracked)
+            """,
+        )
+
+        launch_command('anno', '-b=feature', 'Custom annotation for `feature` branch')
+        assert_command(
+            ["status"],
+            """
+            master (untracked)
+
+            develop *  Custom annotation for `develop` branch (untracked)
+            |
+            x-feature  Custom annotation for `feature` branch (untracked)
+            """,
+        )
+
+        launch_command('anno', '-b=refs/heads/feature', 'Custom annotation for `feature` branch')
```

Comment:
Also! for clarity, what happens if `refs/remote/origin/...` or `refs/tags/...` for an actually existing remote branch/tag is passed? And what happens if a string NOT corresponding to any branch in any way is passed?

I'm mostly concerned whether the error messages are clean (and not some obscure stack traces)

Reply:
@PawelLipski Same error for each case:

**refs/remote/origin/...:**
`➜ git machete anno -b=refs/remote/origin/develop 'test'
Branch refs/remote/origin/develop not found in the tree of branch dependencies.
Use git machete add refs/remote/origin/develop or git machete edit
`

**refs/tags/...:**
`➜ git machete anno -b=refs/tags/v3.9.1 'test'
Branch refs/tags/v3.9.1 not found in the tree of branch dependencies.
Use git machete add refs/tags/v3.9.1 or git machete edit
`

**non-existing branch:**
`➜ git machete anno -b=random_branch 'test'   
Branch random_branch not found in the tree of branch dependencies.
Use git machete add random_branch or git machete edit
`

Reply:
Gr8 ❤️